drkafka/src/main/java/com/pinterest/doctorkafka/DoctorKafkaMain.java

```diff
@@ -1,128 +1,116 @@
+import com.pinterest.doctorkafka.config.DoctorKafkaAppConfig;
+import com.pinterest.doctorkafka.config.DoctorKafkaConfig;
+import com.pinterest.doctorkafka.replicastats.ReplicaStatsManager;
+import com.pinterest.doctorkafka.servlet.ClusterInfoServlet;
+import com.pinterest.doctorkafka.servlet.DoctorKafkaActionsServlet;
+import com.pinterest.doctorkafka.servlet.DoctorKafkaBrokerStatsServlet;
+import com.pinterest.doctorkafka.servlet.DoctorKafkaInfoServlet;
+import com.pinterest.doctorkafka.servlet.KafkaTopicStatsServlet;
+import com.pinterest.doctorkafka.servlet.UnderReplicatedPartitionsServlet;
+import com.pinterest.doctorkafka.util.OperatorUtil;
+
+import io.dropwizard.Application;
+import io.dropwizard.assets.AssetsBundle;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
 
 /**
- *  DoctorKafka is the central service for managing kafka operation.
+ * DoctorKafka is the central service for managing kafka operation.
  *
  */
-public class DoctorKafkaMain {
-
-  private static final Logger LOG = LogManager.getLogger(DoctorKafkaMain.class);
-  private static final String CONFIG_PATH = "config";
-  private static final String METRICS_TOPIC = "topic";
-  private static final String OSTRICH_PORT = "ostrichport";
-  private static final String TSD_HOSTPORT = "tsdhostport";
-  private static final String UPTIME_IN_SECONDS = "uptimeinseconds";
-  private static final String ZOOKEEPER = "zookeeper";
-  private static final Options options = new Options();
-
-  public static  DoctorKafka doctorKafka = null;
-  private static DoctorKafkaWatcher operatorWatcher = null;
-
-  /**
-   *  Usage:  DoctorKafkaMain  --config config_file_path
-   */
-  private static CommandLine parseCommandLine(String[] args) {
-    Option configPath = new Option(CONFIG_PATH, true, "config file path");
-    Option zookeeper = new Option(ZOOKEEPER, true, "zk url for metrics topic");
-    zookeeper.setRequired(false);
-    Option topic = new Option(METRICS_TOPIC, true, "kafka topic for metric messages");
-    topic.setRequired(false);
-    Option tsdHostPort = new Option(TSD_HOSTPORT, true, "tsd host and port, e.g. localhost:18621");
-    tsdHostPort.setRequired(false);
-    Option ostrichPort = new Option(OSTRICH_PORT, true, "ostrich port");
-    ostrichPort.setRequired(false);
-    Option uptimeInSeconds = new Option(UPTIME_IN_SECONDS, true, "uptime in seconds");
-    uptimeInSeconds.setRequired(false);
-    options.addOption(configPath).addOption(zookeeper).addOption(topic)
-        .addOption(tsdHostPort).addOption(ostrichPort).addOption(uptimeInSeconds);
-
-    if (args.length < 1) {
-      printUsageAndExit();
-    }
-    CommandLineParser parser = new DefaultParser();
-    CommandLine cmd = null;
-    try {
-      cmd = parser.parse(options, args);
-    } catch (ParseException | NumberFormatException e) {
-      printUsageAndExit();
+public class DoctorKafkaMain extends Application<DoctorKafkaAppConfig> {
```

Comment:
can we avoid changing the application starting method in this change?  Is this related to using dropwizard to allow rest api development?

Reply:
Unfortunately dropwizard has a standard launch mechanism which will change this. We can mimic it but it's not a good design pattern.

Reply:
Dropwizard is lightweight and fairly popular across the industry. Spring is very heavy due to it's dependency graph. This PR will allow us to support all 3 models:
1. Web UI
2. Servlets
3. REST (jax-rs)